### DISCUSSION TITLE
Lesson 28: Identifier not found or not unique: Vm.Log[]

### DISCUSSION THREAD

**Discussion by ayushgupta0610** [Cyfrin/foundry-full-course-cu#1634] [Q&A]
Is anyone else facing this issue as well? All the tests and every other vm.something is working but for some reason Vm.Log[] isn't. Ran foundryup as well just in case.

**Comment by drained69** [comment]
The problem most likely is what you are trying to log and not from `Vm.log[]`

**Comment by EngrPips** [comment]
Hello @ayushgupta0610, did you import `Vm` into your test contract? If you didn't, you can do so with the code of the line below 

```javascript
import {Vm} from "forge-std/Vm.sol";
```

**Reply by drained69** [reply]
`Vm` imports with Test.

He should try sharing the file of what he is working on.

**Comment by shoaib-eth** [comment]
@ayushgupta0610 It would be super helpful for us to assist you if you could provide the following:

* **The Relevant Code:** Can you share the portion of your test contract where you're using `Vm.Log[]`, including the imports?
* **Exact Error:** What's the full error message you're seeing?
* **What to Log:** Could you explain what kind of data you're intending to log with `Vm.Log[]`?

Also, try these debugging tips:

* **Add `console.log`:**  Insert `console.log` statements before and after your `Vm.Log[]` usage to see the values of relevant variables.
* **Test with Minimal Example:** See if you can reproduce the issue with a very simple test and a basic `Vm.Log[]` call.

Let us know how it goes!
